chihacknight/decarbonize-my-state
# Graph Power Sources

## Overview

Adds a new pie chart showing how power is generated in all states with a % of emissions from power > 0 (so this graph doesn't show for Vermont or DC).

**Note:** I also dropped our recommendation on adding solar to your home - I think that makes more sense in the Take Action page and broke out the flow of the state details page - the main thing is to close down fossil fuel plants, not to get everyone to add solar panels to their roof.

### Demo

| Desktop | Mobile |
| --- | --- |
| ![Screenshot from 2023-03-13 00-16-48](https://user-images.githubusercontent.com/3187531/224614070-4be40727-eacf-4bce-baa0-bb3cdf517d52.png) | ![Screenshot from 2023-03-13 00-17-00](https://user-images.githubusercontent.com/3187531/224614066-c759f542-794f-42cf-aeb5-91cdd86405e2.png) |

### Notes

Holy moly, this one is challenging to handle because of the wide variety of data ranges - Nevada in particular is a great test case because it has a bunch of small slices that are at risk of having overlapping labels.

## Testing Instructions

- Test a variety of states on desktop and mobile and confirm you can read all the labels
- Also confirm the homepage state emissions map tooltip is properly positioned again, particularly for DC and L shaped states like Louisiana
